### ISSUE TITLE
Confusing wording in docs

### ISSUE THREAD

**Issue by cagatay-y** [mesonbuild/meson#1833]
In manual under section [Syntax > Variables](http://mesonbuild.com/Syntax.html#variables) it says "all variables are immutable" and then gives the following example:
```
var1 = [1, 2, 3]
var2 = var1
var2 += [4]
# var2 is now [1, 2, 3, 4]
# var1 is still [1, 2, 3]
```
In Wikipedia and SO questions, I have read that an object is mutable if it can't be changed, but 3rd line changes the object. I think a better way to explain the behaviour above would be to say "assigning a variable to another one makes a deep copy of the copied one." What do you think?

**Comment by jpakkane** [comment]
It would be better to say that this might seem like changing the value of `var2` but that `+=` is just notational shorthand for this:

    var2 = var2 + [4]

That is, if anyone was holding on to the old version of `var2`, they still have it, they don't see the `4` added here. This should be equivalent to value semantics as described, but probably a bit simpler to understand.

**Comment by eli-schwartz** [comment]
The documentation was fixed in #8697
